src/lib/variants.ts

```diff
@@ -152,6 +153,24 @@ export const withinMaxViews = (log: ViewLog, now: Date = new Date()): Filter =>
     },
 });
 
+export const withinArticleViewedSettings = (history: WeeklyArticleHistory): Filter => ({
+    id: 'withinArticleViewedSettings',
+    test: (test, _): boolean => {
+        // Allow test to pass if no articles viewed settings have been set
+        if (!test.articlesViewedSettings || !test.articlesViewedSettings.periodInWeeks) {
+            return true;
+        }
+
+        const { minViews, maxViews, periodInWeeks } = test.articlesViewedSettings;
+
+        const viewCountForWeeks = getArticleViewCountForWeeks(history, periodInWeeks);
+        const minViewsOk = minViews ? viewCountForWeeks >= minViews : true;
+        const maxViewsOk = maxViews ? viewCountForWeeks <= maxViews : true;
```

Comment:
This is probably overthinking it, but would 0 be a valid `maxViews` configuration? I.e. only show this epic if they've never seen an article before (new reader). Would it be more correct to test for null/undefined rather than truthiness? (Since 0 is falsey).

Reply:
Glad you picked up on this because I did think about it and wasn't sure either. I wanted to use `maxViews !== undefined ? ...` as you say but didn't because:
1) Makes the ternary more confusing to read (somebody will always ask if you couldn't just check truthty/falsy which is a more common pattern, although obviously not always the correct one)
2) I couldn't find a use case for either `minViews` or `maxViews` to be zero; if the tests don't apply rules on this, the containing object will be empty and therefore these values will be `undefined` not `0` (I think in your use case of new users only, `maxViews` would have to be `1` and `minViews` would have to be `0` in which case it will fall back to `true` which is fine).
3) This is how Frontend checks the values, so we'd only want to change it if there's a clear reason